test/versioned/elastic/elasticsearch.tap.js

```diff
@@ -0,0 +1,378 @@
+      t.match(attrs.host, METRIC_HOST_NAME)
+      transaction.end()
+      t.ok(agent.queries.samples.size > 0, 'there should be a query sample')
+      for (const query of agent.queries.samples.values()) {
+        t.ok(query.total > 0, 'the samples should have positive duration')
+        t.match(
+          query.trace.query,
+          JSON.stringify(expectedQuery),
+          'expected msearch query should have been recorded'
+        )
+      }
+    })
+  })
+
+  t.test('should create correct metrics', async function (t) {
+    t.plan(28)
+    await helper.runInTransaction(agent, async function transactionInScope() {
+      const transaction = agent.getTransaction()
+
+      try {
+        await client.index({
+          index: DB_INDEX,
+          id: 'testkey2',
+          document: {
+            title: 'second document',
+            body: 'body of the second document'
+          }
+        })
+      } catch (e) {
+        t.notOk(e, 'Create document should not error')
+      }
+
+      // check metrics/methods for "exists" queries
+      await client.exists({ id: 'testkey2', index: DB_INDEX })
+      await client.get({ id: 'testkey2', index: DB_INDEX })
+      await client.search({ query: { match: { body: 'document' } } })
+      await client.delete({ id: 'testkey2', index: DB_INDEX })
+      transaction.end()
+
+      const unscoped = transaction.metrics.unscoped
+      const expected = {
+        'Datastore/all': 5,
+        'Datastore/allWeb': 5,
+        'Datastore/ElasticSearch/all': 5,
+        'Datastore/ElasticSearch/allWeb': 5,
+        'Datastore/operation/ElasticSearch/doc.update': 1,
+        'Datastore/operation/ElasticSearch/doc.get': 1,
+        'Datastore/operation/ElasticSearch/doc.exists': 1,
+        'Datastore/operation/ElasticSearch/search': 1,
+        'Datastore/statement/ElasticSearch/test/doc.update': 1,
+        'Datastore/statement/ElasticSearch/test/doc.get': 1,
+        'Datastore/statement/ElasticSearch/test/doc.exists': 1,
+        'Datastore/statement/ElasticSearch/test/doc.delete': 1,
+        'Datastore/statement/ElasticSearch/any/search': 1
+      }
+      expected['Datastore/instance/ElasticSearch/' + HOST_ID] = 5
+      checkMetrics(t, unscoped, expected)
+    })
+  })
+
+  t.test('should not add instance attributes/metrics when disabled', async function (t) {
+    t.plan(4)
+
+    // disable
+    agent.config.datastore_tracer.instance_reporting.enabled = false
+    agent.config.datastore_tracer.database_name_reporting.enabled = false
+
+    await helper.runInTransaction(agent, async function transactionInScope() {
+      const transaction = agent.getTransaction()
+
+      try {
+        await client.index({
+          index: DB_INDEX,
+          id: 'testkey3',
+          document: {
+            title: 'third document title',
+            body: 'body of the third document'
+          }
+        })
+      } catch (e) {
+        t.notOk(e, 'Create document should not error')
```

Comment:
I don't think you need these try/catch. if an async call fails, the test will fail.